### PR TITLE
fix: match boolean null for substrait

### PR DESCRIPTION
**Pull request by waynexia**
# Which issue does this PR close?

Closes #.

# Rationale for this change

Forget to match boolean null in #5775. And fix a test from #5946 

# What changes are included in this PR?

- support boolean null literal
- fix test runner `round_trip_literal`

# Are these changes tested?

yes

# Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/substrait/src/logical_plan/producer.rs [hunk 1]
```diff
@@ -1330,9 +1335,7 @@ mod test {
     fn round_trip_literal(scalar: ScalarValue) -> Result<()> {
         println!("Checking round trip of {:?}", scalar);
 
-        let scalar = ScalarValue::Int32(Some(i32::MAX));
```
Comment (alamb) [inline datafusion/substrait/src/logical_plan/producer.rs:1333]: so embarrassing 🤦

**Review by jackwener — APPROVED** [review]
Thank you @waynexia

**Review by alamb — APPROVED** [review]
Thanks @waynexia

File: datafusion/substrait/src/logical_plan/producer.rs (hunk 1 above)
Comment (waynexia) [reply datafusion/substrait/src/logical_plan/producer.rs:1333]: that's a part of being human :P

Comment (alamb) [reply datafusion/substrait/src/logical_plan/producer.rs:1333]: 🤖  lol

**Pull request merged**